### PR TITLE
Feature/donation count and duration state

### PR DESCRIPTION
**Pull request by jekuaitk**
* Keeps total donation count and duration in state
* Updates state when a new donation is made
* Disables cache on front-page

### CODE REVIEW

File: web/modules/custom/giv_din_stemme/src/Helper/Helper.php [hunk 1]
```diff
@@ -37,13 +60,78 @@ public function __construct(
...
+    $this->getLock();
+
+    $totalDonationDuration = $this->getTotalDonationDuration();
+
+    if (is_null($totalDonationDuration)) {
+      $this->state->set(self::GIV_DIN_STEMME_TOTAL_DONATION_DURATION_STATE_KEY, $duration);
+    }
+    else {
+      $this->state->set(self::GIV_DIN_STEMME_TOTAL_DONATION_DURATION_STATE_KEY, $totalDonationDuration + $duration);
+    }
```
Comment (rimi-itk) [inline web/modules/custom/giv_din_stemme/src/Helper/Helper.php:92]:
```suggestion
    $totalDonationDuration = $this->getTotalDonationDuration();

    $this->state->set(self::GIV_DIN_STEMME_TOTAL_DONATION_DURATION_STATE_KEY, ($totalDonationDuration ?? 0) + $duration);
```

or

```suggestion
    $totalDonationDuration = $this->getTotalDonationDuration() ?? 0;
    $this->state->set(self::GIV_DIN_STEMME_TOTAL_DONATION_DURATION_STATE_KEY, $totalDonationDuration + $duration);
```

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: web/modules/custom/giv_din_stemme/src/Helper/Helper.php [hunk 2]
```diff
@@ -37,13 +60,78 @@ public function __construct(
...
+
+  /**
+   * Gets the GIV_DIN_STEMME_UPDATE_STATE_LOCK lock.
+   */
+  private function getLock(): void {
+    while (!$this->lock->acquire(self::GIV_DIN_STEMME_UPDATE_STATE_LOCK)) {
+      // @see https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Lock%21LockBackendAbstract.php/function/LockBackendAbstract%3A%3Await/11.x
+      $this->lock->wait(self::GIV_DIN_STEMME_UPDATE_STATE_LOCK);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (jekuaitk) [reply web/modules/custom/giv_din_stemme/src/Helper/Helper.php:133]: Changed reference link to https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Lock%21LockBackendInterface.php/group/lock/11.x, in which it explains acquire/wait.

File: web/themes/custom/giv_din_stemme_theme/texts/consent-text.html
[the remark this answers is not in the document]
Comment (jekuaitk) [reply web/themes/custom/giv_din_stemme_theme/texts/consent-text.html:1]: Nope, but lets fix it here anyways.

File: web/modules/custom/giv_din_stemme/src/Helper/Helper.php [hunk 3]
```diff
@@ -112,12 +106,7 @@ public function updateTotalNumberOfDonations(): void {
 
     $currentTotal = $this->getTotalNumberOfDonations();
 
-    if (is_null($currentTotal)) {
-      $this->state->set(self::GIV_DIN_STEMME_DONATION_COUNT_STATE_KEY, 1);
-    }
-    else {
-      $this->state->set(self::GIV_DIN_STEMME_DONATION_COUNT_STATE_KEY, $currentTotal + 1);
-    }
+    $this->state->set(self::GIV_DIN_STEMME_DONATION_COUNT_STATE_KEY, ($currentTotal ?? 0) + 1);
```
Comment (rimi-itk) [inline web/modules/custom/giv_din_stemme/src/Helper/Helper.php:109]: Handle this like you handle the total above.

File: web/modules/custom/giv_din_stemme/src/Helper/Helper.php [hunk 4]
```diff
@@ -126,8 +115,9 @@ public function updateTotalNumberOfDonations(): void {
    * Gets the GIV_DIN_STEMME_UPDATE_STATE_LOCK lock.
    */
   private function getLock(): void {
+    // Attempt acquiring lock, wait if failed.
+    // @see https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Lock%21LockBackendInterface.php/group/lock/11.x
```
Comment (rimi-itk) [inline web/modules/custom/giv_din_stemme/src/Helper/Helper.php:119]: Thank you! 🎉

**Review by rimi-itk — CHANGES_REQUESTED** [review]

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**